crates/floresta-wire/src/p2p_wire/tests/utils.rs

```diff
@@ -98,7 +98,36 @@ impl TestPeer {
             let req = self.node_rx.recv().await.unwrap();
```

Comment:
Here it is giving me a `RecvError` like this: ``Result::unwrap()` on an `Err` value: RecvError`, I donn't know why, probably because I was not using the `unsafe` rust component in the `setup_node` function because `.run` function in `running_node` consumed a `mut self` instead of `&mut self`. 

So I dont know what to do here now, please help @Davidson-Souza .

Reply:
It didn't do throw this `RecvError`, when I was doing the same thing with chain_selector in #180. Here that unsafe rust component is used and the node is executed in `futures::timeout`.

Reply:
`RecvError` is usually caused by the other end of this channel being dropped. Do you call `read` after the node dies?

Reply:
Like its giving `ReceivError` just after the ChainSelector starts and the peer sends `Peer::Ready`. The debugging looked something like this: 

```
RUNNING NODE
CHAIN SELECTOR
FROM PEER: READY
thread 'async-std/runtime' panicked at crates/floresta-wire/src/p2p_wire/tests/utils.rs:99:49:
called `Result::unwrap()` on an `Err` value: RecvError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
REQUESTING BLOCKS

```

It doesn't even run one complete loop of `Chain_Selector::run()` 

It threw this error for the first time this line was called: 
``` 
let req = self.node_rx.recv().await.unwrap();
```
probably this means, the channel was dropped as soon as it was created. Maybe you can use some different technique to set up the running node.
@Davidson-Souza

Reply:
I think I found it. The problem seems be caused by the node creating connections by itself outside our mock peers and ends-up in a corrupted state. This causes the node to send a message to an uninitialized peer, hitting an `assert!` inside `Peer`. Once the thread panics, apparently the node task dies too. That's why the `Sender` is being dropped. 

```
thread 'async-std/runtime' panicked at crates/floresta-wire/src/p2p_wire/tests/utils.rs:98:49:
called `Result::unwrap()` on an `Err` value: RecvError

---------- This causes all the trouble, and happens because node sends a request to an uninitialized peer ----------
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'async-std/runtime' panicked at crates/floresta-wire/src/p2p_wire/peer.rs:213:9:
assertion `left == right` failed
  left: SentVersion(Instant { tv_sec: 512644, tv_nsec: 11497069 })
 right: Connected
```

The easiest way to solve this is by using `Regtest` insted of `Signet` because it has no addresses for `florestad` to connect to.